dotCMS/src/integration-test/java/com/dotmarketing/portlets/variant/business/VariantAPITest.java

```diff
@@ -0,0 +1,371 @@
+     * Should: throw {@link NullPointerException}
+     *
+     * @throws DotDataException
+     */
+    @Test(expected = IllegalArgumentException.class)
+    public void saveWithoutName() throws DotDataException {
+        final Variant variant = new Variant("1", null, false);
+        APILocator.getVariantAPI().save(variant);
+    }
+
+    /**
+     * Method to test: {@link VariantFactory#save(Variant)}
+     * When: Try to save a archived {@link Variant} object without name
+     * Should: throw {@link NullPointerException}
+     *
+     * @throws DotDataException
+     */
+    @Test(expected = IllegalArgumentException.class)
+    public void saveArchive() throws DotDataException {
+        final Variant variant = new VariantDataGen().archived(true).next();
+        APILocator.getVariantAPI().save(variant);
+    }
+
+    /**
+     * Method to test: {@link VariantFactory#update(Variant)}
+     * When: Try to update a {@link Variant} object
+     * Should: Update it in Data base.
+     *
+     * @throws DotDataException
+     */
+    @Test
+    public void update() throws DotDataException {
+        final Variant variant = new VariantDataGen().nextPersisted();
+
+        assertNotNull(variant);
+        assertNotNull(variant.getIdentifier());
+
+        final Variant variantUpdated = new Variant(variant.getIdentifier(),
+                variant.getName() + "_updated", false);
+
+        APILocator.getVariantAPI().update(variantUpdated);
+
+        final Variant variantFromDataBase = getVariantFromDataBase(variant);
+
+        assertEquals(variantUpdated.getName(), variantFromDataBase.getName());
+        assertEquals(variantUpdated.getIdentifier(), variantFromDataBase.getIdentifier());
+        assertFalse(variantFromDataBase.isArchived());
+    }
+
+    /**
+     * Method to test: {@link VariantFactory#update(Variant)}
+     * When: Try to update a {@link Variant} object that not exists
+     * Should: thorw a {@link DoesNotExistException}
+     *
+     * @throws DotDataException
+     */
+    @Test(expected = DoesNotExistException.class)
+    public void updateNotExists() throws DotDataException {
+        final Variant variantToUpdated = new VariantDataGen()
+                .id("Not_Exists").next();
+
+        APILocator.getVariantAPI().update(variantToUpdated);
+    }
+
+    /**
+     * Method to test: {@link VariantFactory#save(Variant)}
+     * When: Try to update a {@link Variant} object without name
+     * Should: throw {@link NullPointerException}
+     *
+     * @throws DotDataException
+     */
+    @Test(expected = IllegalArgumentException.class)
+    public void updateWithoutName() throws DotDataException {
+        final Variant variant = new Variant("1", null, false);
+        APILocator.getVariantAPI().update(variant);
+    }
+
+    /**
+     * Method to test: {@link VariantFactory#save(Variant)}
+     * When: Try to update a {@link Variant} object without id
+     * Should: throw {@link NullPointerException}
```

Comment:
We don't need this test anymore because We are using Immutable now and it does not allow to create the Variant with a null name

https://github.com/dotCMS/core/pull/22839/commits/8aaaf9e8f7d06c71edfd820737a9fcabc12d94ed#diff-8e7f19b3642200234b515975babfc36e25fa2da824d7b533b2c3ec26a350d100L120